test/InterchainTokenFactory.js

```diff
@@ -162,6 +163,77 @@ describe('InterchainTokenFactory', () => {
         });
     });
 
+    describe('Gateway Interchain Token Factory', async () => {
+        let token, tokenId, tokenManagerAddress, symbol;
+        const tokenCap = BigInt(1e18);
+
+        async function deployToken(salt, lockUnlock = true) {
+            let tokenAddress = AddressZero;
+
+            if (lockUnlock) {
+                token = await deployContract(wallet, 'TestInterchainTokenStandard', [
+                    name,
+                    symbol,
+                    decimals,
+                    service.address,
+                    getRandomBytes32(),
+                ]);
+                tokenAddress = token.address;
+            }
+
+            const params = defaultAbiCoder.encode(
+                ['string', 'string', 'uint8', 'uint256', 'address', 'uint256'],
+                [name, symbol, decimals, 0, tokenAddress, 0],
+            );
+
+            await (await gateway.deployToken(params, getRandomBytes32())).wait();
+
+            tokenId = await service.interchainTokenId(AddressZero, salt);
+            tokenManagerAddress = await service.tokenManagerAddress(tokenId);
+
+            if (lockUnlock) {
+                await (await token.mint(wallet.address, tokenCap)).wait();
```

Comment:
use await then syntax for consistency everywhere

Reply:
done